### PR TITLE
000 - Added NodeJS Github Actions

### PR DESCRIPTION
**Pull request by Adnanhoq**
Added NodeJS build

### CODE REVIEW

File: .github/workflows/NodeJS.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
```
Comment (TomBillingtonUK) [inline .github/workflows/NodeJS.yml:6]: Can you set this trigger as well, please -> on: workflow_dispatch (Sam is trying on BE to do the same)

File: .github/workflows/NodeJS.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x, 20.x, 22.x]
```
Comment (TomBillingtonUK) [inline .github/workflows/NodeJS.yml:19]: you only need to run it on one node version

File: .github/workflows/NodeJS.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - run: npm ci
+    - run: npm run build --if-present
```
Comment (JennicaMistry) [inline .github/workflows/NodeJS.yml:31]: You need a build script.

**Review by JennicaMistry — CHANGES_REQUESTED** [review]

File: .github/workflows/NodeJS.yml (hunk 3 above)
Comment (Adnanhoq) [reply .github/workflows/NodeJS.yml:31]: Added it in new PR: https://github.com/Adnanhoq/TheExtraSemicolon_FrontEnd/pull/2

**Review by Leebee1392 — APPROVED** [review]

**Pull request merged**